spec/rubocop/target_finder_spec.rb

```diff
@@ -79,8 +79,172 @@
     create_empty_file('.hidden/ruby4.rb')
   end
 
-  describe '#find' do
-    let(:found_files) { target_finder.find(args) }
+  shared_examples 'common behavior for #find' do
+    context 'when a file with a ruby filename is passed' do
+      let(:args) { RUBY_FILENAMES.map { |name| "dir2/#{name}" } }
+
+      it 'picks all the ruby files' do
+        expect(found_basenames).to eq(RUBY_FILENAMES)
+      end
+    end
+
+    context 'when files with ruby interpreters are passed' do
+      let(:args) { RUBY_INTERPRETERS.map { |name| "dir2/#{name}" } }
+
+      before do
+        RUBY_INTERPRETERS.each do |interpreter|
+          create_file("dir2/#{interpreter}", "#!/usr/bin/#{interpreter}")
+        end
+      end
+
+      it 'picks all the ruby files' do
+        expect(found_basenames).to eq(RUBY_INTERPRETERS)
+      end
+    end
+
+    context 'when a pattern is passed' do
+      let(:args) { ['dir1/*2.rb'] }
+
+      it 'finds files which match the pattern' do
+        expect(found_basenames).to eq(['ruby2.rb'])
+      end
+    end
+
+    context 'when same paths are passed' do
+      let(:args) { %w[dir1 dir1] }
+
+      it 'does not return duplicated file paths' do
+        count = found_basenames.count { |f| f == 'ruby1.rb' }
+        expect(count).to eq(1)
+      end
+    end
+
+    context 'when some paths are specified in the configuration Exclude ' \
+            'and they are explicitly passed as arguments' do
+      before do
+        create_file('.rubocop.yml', <<~YAML)
+          AllCops:
+            Exclude:
+              - dir1/ruby1.rb
+              - 'dir2/*'
+        YAML
+
+        create_file('dir1/.rubocop.yml', <<~YAML)
+          AllCops:
+            Exclude:
+              - executable
+        YAML
+      end
+
+      let(:args) do
+        ['dir1/ruby1.rb', 'dir1/ruby2.rb', 'dir1/exe*', 'dir2/ruby3.rb']
+      end
+
+      context 'normally' do
+        it 'does not exclude them' do
+          expect(found_basenames)
+            .to eq(['ruby1.rb', 'ruby2.rb', 'executable', 'ruby3.rb'])
+        end
+      end
+
+      context "when it's forced to adhere file exclusion configuration" do
+        let(:force_exclusion) { true }
+
+        it 'excludes them' do
+          expect(found_basenames).to eq(['ruby2.rb'])
+        end
+      end
+    end
+
+    it 'returns absolute paths' do
+      expect(found_files.empty?).to be(false)
+      found_files.each do |file|
+        expect(file.sub(/^[A-Z]:/, '')).to start_with('/')
```

Comment:
This test fails on Windows.

```console
  1) RuboCop::TargetFinder#find(..., :all_file_types) returns absolute paths
     Failure/Error: expect(file.sub(/^[A-Z]:/, '')).to start_with('/')
       expected "d:/a/_temp/d20200504-1968-jbdj8j/work/dir1/executable" to start with "/"
     Shared Example Group: "common behavior for #find" called from ./spec/rubocop/target_finder_spec.rb:433
     # ./spec/rubocop/target_finder_spec.rb:162:in `block (4 levels) in <top (required)>'
     # ./spec/rubocop/target_finder_spec.rb:161:in `each'
     # ./spec/rubocop/target_finder_spec.rb:161:in `block (3 levels) in <top (required)>'
     # ./lib/rubocop/rspec/shared_contexts.rb:29:in `block (4 levels) in <top (required)>'
     # ./lib/rubocop/path_util.rb:67:in `chdir'
     # ./lib/rubocop/path_util.rb:67:in `chdir'
     # ./lib/rubocop/rspec/shared_contexts.rb:28:in `block (3 levels) in <top (required)>'
     # ./lib/rubocop/rspec/shared_contexts.rb:8:in `block (2 levels) in <top (required)>'
     # tasks/spec_runner.rake:29:in `block in run_specs'
     # tasks/spec_runner.rake:41:in `with_encoding'
     # tasks/spec_runner.rake:25:in `run_specs'
     # tasks/spec_runner.rake:112:in `block in <top (required)>'
```

https://github.com/rubocop-hq/rubocop/pull/7930/checks?check_run_id=643778689#step:6:214

Reply:
Thanks! I forgot to check the builds. I'll fix it.